### DISCUSSION TITLE
Manually schedule job from another application.

### DISCUSSION THREAD

**Discussion by ashwin47** [bensheldon/good_job#1281] [Q&A] [answered]
I have a Rails application and a Python Discord bot both accessing the same PostgreSQL database. I want to schedule a job from the Python bot. If I add an entry to the "good_job" table, will it be picked up?

**Comment by bensheldon** [comment, marked as answer]
**You can.** New records inserted into the `good_jobs` table will get picked up on polling. You could also [reverse engineer and use NOTIFY too](https://github.com/bensheldon/good_job/blob/3f7817f5a4f0fcecb21c2fd62a1fe80e62673890/lib/good_job/notifier.rb#L55). 

**You shouldn't.** That would end up coupling the two services via the database, which would make upgrades and deployment more difficult. Also you'd need to manage both GoodJob updates and Rails updates may modify the contents of `serialized_params` (how Active Job serializes the job data). It would be hard.

Instead, I'd recommend making a simple HTTP API with a shared secret, or using [Twirp](https://github.com/twitchtv/twirp)/gRPC.
